FRONTEND/src/modules.js/navBar.js

```diff
@@ -0,0 +1,354 @@
+    var barSize = canvasActualWidth / numberOfBars;
+
+    for (categ in this.options.data) {
+      var val = this.options.data[categ];
+      var barHeight = Math.round((canvasActualHeight * val) / maxValue);
+      drawBar(
+        this.ctx,
+        this.options.padding + barIndex * barSize,
+        this.canvas.height - barHeight - this.options.padding,
+        barSize,
+        barHeight,
+        this.colors[barIndex % this.colors.length]
+      );
+
+      barIndex++;
+    }
+  };
+};
+
+
+
+// ------------------------------ Function-5 ------------------------------
+// Function to scale the horitzontal values of the range slider
+function logslider(position) {
+    // position will be between 0 and 100
+    var minp = 0;
+    var maxp = 100;
+  
+    // The result should be between 100 an 10000000
+    var minv = Math.log(parseInt(Object.keys(OccurData)[0]));
+    var maxv = Math.log(
+      parseInt(Object.keys(OccurData)[Object.keys(OccurData).length - 1])
+    );
+  
+    // calculate adjustment factor
+    var scale = (maxv - minv) / (maxp - minp);
+  
+    return Math.trunc(Math.exp(minv + scale * (position - minp)));
+  }
+
+
+
+// ------------------------------ Function-6 ------------------------------
+function sliderRangeFunction() {
+    $("#year-slider-range").slider({
+        range: true,
+        min: parseInt(Object.keys(YearData)[0]),
+        max: parseInt(Object.keys(YearData)[Object.keys(YearData).length - 1]),
+        values: [
+            parseInt(Object.keys(YearData)[0]),
+            parseInt(Object.keys(YearData)[Object.keys(YearData).length - 1]),
+        ],
+        slide: function (event, ui) {
+            $("#yearAmount").val(ui.values[0] + " - " + ui.values[1]);
+        },
+        change: function () {
+            updateNodes();
+        },
+        create: function () {
+            $("#yearAmount").val(
+                $("#year-slider-range").slider("values", 0) +
+                " - " +
+                $("#year-slider-range").slider("values", 1)
+            );
+        },
+    });
+    $("#occur-slider-range").slider({
+        range: true,
+        min: 0,
+        max: 100,
+        values: [20, 100],
+        slide: function (event, ui) {
+            $("#occurAmount").val(
+                logslider(ui.values[0]) + " - " + logslider(ui.values[1])
+            );
+        },
+        change: function () {
+            updateNodes();
+        },
+        create: function () {
+            $("#occurAmount").val(
+                logslider($("#occur-slider-range").slider("values", 0)) +
+                " - " +
+                logslider($("#occur-slider-range").slider("values", 1))
+            );
+        },
+    });
+};
+
+
+
+// ------------------------------ Function-7 ------------------------------
+// Exportamos la función de inicialización del autocompletado
+function initAutocomplete(toolTopicData, addNodesFn, toolImage, databaseImage, topicImage) {
+    $(function () {
```

Comment:
Do you really need this jquery init here again?

Reply:
PR Request solved. Eliminate unnecessary jquery of function initAutocomplete of FRONTEND/src/modules.js/navBar.js